### PR TITLE
Fix: clean up the created namespace when Create() failed

### LINKED ISSUE #216: Bug(helm-style plugins): If the create() fails, we need to clean up the created namespace
## Describe the Bug

If the create() fails, we need to clean up the created namespace.

```sh
2022-02-22 21:10:36 ℹ [INFO]  Processing: jenkins-dev -> Create ...
2022-02-22 21:10:36 ℹ [DEBUG]  Start: 

2022-02-22 21:10:36 ✖ [ERROR]  Plugin jenkins-dev Create failed.
2022-02-22 21:10:36 ℹ [DEBUG]  End:

2022-02-22 21:10:36 ℹ [INFO]  jenkins-dev-Create -> namespaces "jenkins" already exists
2022-02-22 21:10:36 ℹ [INFO]  handle-result -> plugin jenkins-dev Create failed
2022-02-22 21:10:36 ✖ [ERROR]  Apply failed => some error(s) occurred during plugins apply process.
```

## How to Reproduce

```yaml
---
tools:
  - name: jenkins-dev
    plugin:
      kind: jenkins
      version: 0.0.2
    options:
      create_namespace: true
      repo:
        name: jenkins
        url: https://charts.jenkins.io
      chart:
        chart_name: jenkins/jenkins
        release_name: dev
        namespace: jenkins
        wait: true
        timeout: 5m
        upgradeCRDs: true

```

### PR DESCRIPTION
**Pull request by daniel-hutao**
Signed-off-by: Daniel Hu <tao.hu@merico.dev>

# Summary

- clean up the created namespace when Create() failed
- capitalize the log output

### Related Issues

fix #216

### CODE REVIEW

File: cmd/devstream/verify.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ func verifyCMDFunc(cmd *cobra.Command, args []string) {
 	}
 
 	if healthy {
-		log.Success("all tools are healthy")
+		log.Success("All tools are healthy")
```
Comment (IronCore864) [inline cmd/devstream/verify.go:26]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/argocd/create.go [hunk 2]
```diff
@@ -44,15 +50,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+	return nil
+}
+
+// TODO(daniel-hutao): All helm-style plugins has this code logic, maybe it's better to move it to a common package.
+func dealWithNsWhenInterruption(param *Param) error {
+	if !param.CreateNamespace {
+		return nil
+	}
+
+	log.Debugf("Prepare to delete the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/argocd/create.go:82]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/argocd/create.go [hunk 3]
```diff
@@ -44,15 +50,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+	log.Debugf("Prepare to create the namespace: %s", param.Chart.Namespace)
+
+	kubeClient, err := k8s.NewClient()
+	if err != nil {
+		return err
+	}
+
+	err = kubeClient.CreateNamespace(param.Chart.Namespace)
+	if err != nil {
+		log.Debugf("Failed to create the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/argocd/create.go:68]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/argocd/create.go [hunk 4]
```diff
@@ -44,15 +50,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+
 	kubeClient, err := k8s.NewClient()
 	if err != nil {
 		return err
 	}
 
-	return kubeClient.CreateNamespace(param.Chart.Namespace)
+	err = kubeClient.DeleteNamespace(param.Chart.Namespace)
+	if err != nil {
+		log.Debugf("Failed to delete the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/argocd/create.go:91]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/argocd/delete.go [hunk 5]
```diff
@@ -28,7 +28,7 @@ func Delete(options *map[string]interface{}) (bool, error) {
 		return false, err
 	}
 
-	log.Info("uninstalling argocd helm chart")
+	log.Info("Uninstalling argocd helm chart")
```
Comment (IronCore864) [inline internal/pkg/plugin/argocd/delete.go:31]: Missing "period" (.) at the end of the sentence.

Uninstalling -> Deleting (to be consistent with our interface's name)

File: internal/pkg/plugin/githubactions/golang/update.go [hunk 6]
```diff
@@ -25,7 +25,7 @@ func Update(options *map[string]interface{}) (map[string]interface{}, error) {
 		return nil, err
 	}
 
-	log.Infof("language is %s", ga.GetLanguage(opt.Language))
+	log.Infof("Language is %s", ga.GetLanguage(opt.Language))
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/update.go:28]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/githubactions/nodejs/delete.go [hunk 7]
```diff
@@ -23,7 +23,7 @@ func Delete(options *map[string]interface{}) (bool, error) {
 		return false, err
 	}
 
-	log.Infof("language is %s", ga.GetLanguage(opt.Language))
+	log.Infof("Language is %s", ga.GetLanguage(opt.Language))
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/nodejs/delete.go:26]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/githubactions/python/delete.go [hunk 8]
```diff
@@ -23,7 +23,7 @@ func Delete(options *map[string]interface{}) (bool, error) {
 		return false, err
 	}
 
-	log.Infof("language is %s", ga.GetLanguage(opt.Language))
+	log.Infof("Language is %s", ga.GetLanguage(opt.Language))
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/python/delete.go:26]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/githubactions/python/update.go [hunk 9]
```diff
@@ -23,7 +23,7 @@ func Update(options *map[string]interface{}) (map[string]interface{}, error) {
 		return nil, err
 	}
 
-	log.Infof("language is %s", ga.GetLanguage(opt.Language))
+	log.Infof("Language is %s", ga.GetLanguage(opt.Language))
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/python/update.go:26]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/jenkins/create.go [hunk 10]
```diff
@@ -43,6 +43,12 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
 
 	log.Info("Installing or updating jenkins helm chart ...")
 	if err = h.InstallOrUpgradeChart(); err != nil {
+		log.Debugf("Failed to install or upgrade the Chart: %s", err)
+		if err = dealWithNsWhenInterruption(&param); err != nil {
+			log.Debugf("Failed to deal with namespace: %s", err)
```
Comment (IronCore864) [inline internal/pkg/plugin/jenkins/create.go:48]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/jenkins/create.go [hunk 11]
```diff
@@ -53,15 +59,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+	log.Debugf("Prepare to create the namespace: %s", param.Chart.Namespace)
+
+	kubeClient, err := k8s.NewClient()
+	if err != nil {
+		return err
+	}
+
+	err = kubeClient.CreateNamespace(param.Chart.Namespace)
+	if err != nil {
+		log.Debugf("Failed to create the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/jenkins/create.go:77]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/jenkins/create.go [hunk 12]
```diff
@@ -53,15 +59,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+	return nil
+}
+
+// TODO(daniel-hutao): All helm-style plugins has this code logic, maybe it's better to move it to a common package.
+func dealWithNsWhenInterruption(param *Param) error {
+	if !param.CreateNamespace {
+		return nil
+	}
+
+	log.Debugf("Prepare to delete the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/jenkins/create.go:91]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/jenkins/create.go [hunk 13]
```diff
@@ -53,15 +59,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+
 	kubeClient, err := k8s.NewClient()
 	if err != nil {
 		return err
 	}
 
-	return kubeClient.CreateNamespace(param.Chart.Namespace)
+	err = kubeClient.DeleteNamespace(param.Chart.Namespace)
+	if err != nil {
+		log.Debugf("Failed to delete the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/jenkins/create.go:100]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/kubeprometheus/create.go [hunk 14]
```diff
@@ -35,6 +35,12 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
 
 	log.Info("Installing or updating kube-prometheus-stack helm chart ...")
 	if err = h.InstallOrUpgradeChart(); err != nil {
+		log.Debugf("Failed to install or upgrade the Chart: %s", err)
+		if err = dealWithNsWhenInterruption(&param); err != nil {
+			log.Debugf("Failed to deal with namespace: %s", err)
```
Comment (IronCore864) [inline internal/pkg/plugin/kubeprometheus/create.go:40]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/kubeprometheus/create.go [hunk 15]
```diff
@@ -45,15 +51,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+	log.Debugf("Prepare to create the namespace: %s", param.Chart.Namespace)
+
+	kubeClient, err := k8s.NewClient()
+	if err != nil {
+		return err
+	}
+
+	err = kubeClient.CreateNamespace(param.Chart.Namespace)
+	if err != nil {
+		log.Debugf("Failed to create the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/kubeprometheus/create.go:69]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/kubeprometheus/create.go [hunk 16]
```diff
@@ -45,15 +51,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+	return nil
+}
+
+// TODO(daniel-hutao): All helm-style plugins has this code logic, maybe it's better to move it to a common package.
+func dealWithNsWhenInterruption(param *Param) error {
+	if !param.CreateNamespace {
+		return nil
+	}
+
+	log.Debugf("Prepare to delete the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/kubeprometheus/create.go:83]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/kubeprometheus/create.go [hunk 17]
```diff
@@ -45,15 +51,48 @@ func Create(options *map[string]interface{}) (map[string]interface{}, error) {
...
+
 	kubeClient, err := k8s.NewClient()
 	if err != nil {
 		return err
 	}
 
-	return kubeClient.CreateNamespace(param.Chart.Namespace)
+	err = kubeClient.DeleteNamespace(param.Chart.Namespace)
+	if err != nil {
+		log.Debugf("Failed to delete the namespace: %s", param.Chart.Namespace)
```
Comment (IronCore864) [inline internal/pkg/plugin/kubeprometheus/create.go:92]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/reposcaffolding/github/golang/golang.go [hunk 18]
```diff
@@ -130,7 +130,7 @@ func Render(filePath string, param *Param) ([]byte, error) {
 			Owner: param.Owner,
 		},
 	}
-	log.Debugf("filePath: %s", filePath)
+	log.Debugf("FilePath: %s", filePath)
```
Comment (IronCore864) [inline internal/pkg/plugin/reposcaffolding/github/golang/golang.go:133]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/argocd/delete.go (hunk 5 above)
Comment (lfbdev) [inline internal/pkg/plugin/argocd/delete.go:31]: Missing (.) at the end of the sentense.

File: internal/pkg/plugin/reposcaffolding/github/golang/golang.go [hunk 19]
```diff
@@ -142,13 +142,13 @@ func Render(filePath string, param *Param) ([]byte, error) {
 	tpl := template.New("github-repo-scaffolding-golang").Delims("[[", "]]")
 	parsed, err := tpl.Parse(textStr)
 	if err != nil {
-		log.Debugf("template parse file failed: %s", err)
+		log.Debugf("Template parse file failed: %s", err)
```
Comment (IronCore864) [inline internal/pkg/plugin/reposcaffolding/github/golang/golang.go:145]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/reposcaffolding/github/golang/golang.go [hunk 20]
```diff
@@ -142,13 +142,13 @@ func Render(filePath string, param *Param) ([]byte, error) {
...
 		return nil, err
 	}
 
 	var buf bytes.Buffer
 	if err = parsed.Execute(&buf, config); err != nil {
-		log.Debugf("template execute failed: %s", err)
+		log.Debugf("Template execute failed: %s", err)
```
Comment (IronCore864) [inline internal/pkg/plugin/reposcaffolding/github/golang/golang.go:151]: Missing "period" (.) at the end of the sentense.

File: internal/pkg/plugin/trellogithub/update.go [hunk 21]
```diff
@@ -12,7 +12,7 @@ func Update(options *map[string]interface{}) (map[string]interface{}, error) {
 	}
 
 	api := gis.GetApi()
-	log.Infof("api is %s", api.Name)
+	log.Infof("API is %s", api.Name)
```
Comment (IronCore864) [inline internal/pkg/plugin/trellogithub/update.go:15]: Missing "period" (.) at the end of the sentense.

File: pkg/util/github/commit.go [hunk 22]
```diff
@@ -11,7 +11,7 @@ import (
 func (c *Client) GetLastCommit() (*github.RepositoryCommit, error) {
 	commits, _, err := c.Client.Repositories.ListCommits(c.Context, c.Owner, c.Repo, &github.CommitsListOptions{})
 	if err != nil {
-		log.Debugf("failed to get RepositoryCommits: %s", err)
+		log.Debugf("Failed to get RepositoryCommits: %s", err)
```
Comment (IronCore864) [inline pkg/util/github/commit.go:14]: Missing "period" (.) at the end of the sentense.

File: pkg/util/kubectl/main.go [hunk 23]
```diff
@@ -24,8 +24,8 @@ func kubectlAction(action string, filename string) error {
 	cmd := exec.Command("kubectl", action, "-f", filename)
 	cOut, err := cmd.CombinedOutput()
 	if err != nil {
-		log.Errorf("failed to exec: < %s >", cmd.String())
-		log.Errorf("exec logs: < %s >. got error: %s", string(cOut), err)
+		log.Errorf("Failed to exec: < %s >", cmd.String())
```
Comment (IronCore864) [inline pkg/util/kubectl/main.go:27]: Missing "period" (.) at the end of the sentense.

File: pkg/util/kubectl/main.go [hunk 24]
```diff
@@ -24,8 +24,8 @@ func kubectlAction(action string, filename string) error {
...
+		log.Errorf("Exec logs: < %s >. got error: %s", string(cOut), err)
```
Comment (IronCore864) [inline pkg/util/kubectl/main.go:28]: got error -> Got error

Missing "period" (.) at the end of the sentense.

File: pkg/util/zip/zip.go [hunk 25]
```diff
@@ -29,14 +29,14 @@ func UnZip(filePath, targetPath string) error {
 func handleArchiveFiles(targetPath string, files []*zip.File) error {
 	for _, f := range files {
 		filePath := filepath.Join(targetPath, f.Name)
-		log.Debugf("unzipping file -> %s", filePath)
+		log.Debugf("Unzipping file -> %s", filePath)
```
Comment (IronCore864) [inline pkg/util/zip/zip.go:32]: Missing "period" (.) at the end of the sentense.

File: pkg/util/zip/zip.go [hunk 26]
```diff
@@ -29,14 +29,14 @@ func UnZip(filePath, targetPath string) error {
...
 
 		if !strings.HasPrefix(filePath, filepath.Clean(targetPath)+string(os.PathSeparator)) {
 			return fmt.Errorf("invalid file path")
 		}
 
 		if f.FileInfo().IsDir() {
-			log.Debugf("creating directory -> %s", f.FileInfo().Name())
+			log.Debugf("Creating directory -> %s", f.FileInfo().Name())
```
Comment (IronCore864) [inline pkg/util/zip/zip.go:39]: Missing "period" (.) at the end of the sentense.

**Review by IronCore864 — APPROVED** [review]

**Comment by IronCore864** [comment]
Very good PR to fix logging format issues.

**Pull request merged**